go-controller/pkg/ovn/master.go

```diff
@@ -979,6 +980,9 @@ func (oc *DefaultNetworkController) addUpdateNodeEvent(node *kapi.Node, nSyncs *
 
 	// since the nodeSync objects are created knowing if hybridOverlay is enabled this should work
 	if nSyncs.syncHo {
+		if atomic.LoadUint32(&oc.allInitialPodsProcessed) == 0 {
+			return fmt.Errorf("cannot allocate hybrid overlay distributed router ip for nodes until all initial pods are processed")
+		}
 		if err = oc.allocateHybridOverlayDRIP(node); err != nil {
 			errs = append(errs, err)
```

Comment:
you can just check is ErrAlreadyAllocated and then ignore since the other sync already reserved it